Index: api/Controllers/BookController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\n/**\n * Author: \"Emma Parker\"\n * Date: 5/22/22\n * File: BookController.php\n *Description: define the book controller class\n */\n\nnamespace BooksAPI\\Controllers;\n\n\nuse Psr\\Http\\Message\\ServerRequestInterface as Request;\nuse Psr\\Http\\Message\\ResponseInterface as Response;\nuse BooksAPI\\Models\\Book;\nuse BooksAPI\\Controllers\\ControllerHelper as Helper;\nclass BookController {\n    //list all books\n    public function index(Response $response, array $args) : Response {\n        $results = Book::getBooks();\n        return Helper::withJson($response, $results, 200);\n    }\n\n    public function view(Response $response, array $args) : Response {\n        $book_id = $args['book_id'];\n        $results = Book::getBookById($book_id);\n       return Helper::withJson($response, $results, 200);\n    }\n\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/api/Controllers/BookController.php b/api/Controllers/BookController.php
--- a/api/Controllers/BookController.php	(revision 5a58435ec921a67b7528d40df86c0251d32463b8)
+++ b/api/Controllers/BookController.php	(date 1653342525090)
@@ -13,14 +13,16 @@
 use Psr\Http\Message\ResponseInterface as Response;
 use BooksAPI\Models\Book;
 use BooksAPI\Controllers\ControllerHelper as Helper;
+
+
 class BookController {
     //list all books
-    public function index(Response $response, array $args) : Response {
+    public function index(Request $request, Response $response, array $args) : Response {
         $results = Book::getBooks();
         return Helper::withJson($response, $results, 200);
     }
 
-    public function view(Response $response, array $args) : Response {
+    public function view(Request $request, Response $response, array $args) : Response {
         $book_id = $args['book_id'];
         $results = Book::getBookById($book_id);
        return Helper::withJson($response, $results, 200);
Index: api/Controllers/ControllerHelper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\n/**\n * Author: \"Emma Parker\"\n * Date: 5/20/22\n * File: ControllerHelper.php\n *Description:Define controller helper class\n */\nnamespace BooksAPI\\Controllers;\n\nuse Psr\\Http\\Message\\ResponseInterface as Response;\n\nclass ControllerHelper {\n\n    // This method sends a response of data in JSON format along with a status code\n    public static function withJson(Response $response, $data, int $code) : Response {\n        $payload = json_encode($data);\n        $response->getBody()->write($payload);\n        return $response\n            ->withHeader('Content-Type', 'application/json')\n            ->withStatus($code);\n    }\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/api/Controllers/ControllerHelper.php b/api/Controllers/ControllerHelper.php
--- a/api/Controllers/ControllerHelper.php	(revision 5a58435ec921a67b7528d40df86c0251d32463b8)
+++ b/api/Controllers/ControllerHelper.php	(date 1653342584486)
@@ -7,12 +7,13 @@
  */
 namespace BooksAPI\Controllers;
 
+use http\Env\Request;
 use Psr\Http\Message\ResponseInterface as Response;
 
 class ControllerHelper {
 
     // This method sends a response of data in JSON format along with a status code
-    public static function withJson(Response $response, $data, int $code) : Response {
+    public static function withJson(Request $request,Response $response, $data, int $code) : Response {
         $payload = json_encode($data);
         $response->getBody()->write($payload);
         return $response
Index: .idea/dataSources.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<project version=\"4\">\n  <component name=\"DataSourceManagerImpl\" format=\"xml\" multifile-model=\"true\">\n    <data-source source=\"LOCAL\" name=\"books_I425@localhost\" uuid=\"70565d18-468d-4197-94d4-880077ac668d\">\n      <driver-ref>mariadb</driver-ref>\n      <synchronize>true</synchronize>\n      <jdbc-driver>org.mariadb.jdbc.Driver</jdbc-driver>\n      <jdbc-url>jdbc:mariadb://localhost:3306/books_I425</jdbc-url>\n      <working-dir>$ProjectFileDir$</working-dir>\n    </data-source>\n  </component>\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/dataSources.xml b/.idea/dataSources.xml
--- a/.idea/dataSources.xml	(revision 5a58435ec921a67b7528d40df86c0251d32463b8)
+++ b/.idea/dataSources.xml	(date 1653342527881)
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="DataSourceManagerImpl" format="xml" multifile-model="true">
-    <data-source source="LOCAL" name="books_I425@localhost" uuid="70565d18-468d-4197-94d4-880077ac668d">
+    <data-source source="LOCAL" name="group5_books_db@localhost" uuid="f452442c-e51c-4ee1-b56f-4fd50d500961">
       <driver-ref>mariadb</driver-ref>
       <synchronize>true</synchronize>
       <jdbc-driver>org.mariadb.jdbc.Driver</jdbc-driver>
-      <jdbc-url>jdbc:mariadb://localhost:3306/books_I425</jdbc-url>
+      <jdbc-url>jdbc:mariadb://localhost:3306/group5_books_db</jdbc-url>
       <working-dir>$ProjectFileDir$</working-dir>
     </data-source>
   </component>
Index: .idea/dataSources.local.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<project version=\"4\">\n  <component name=\"dataSourceStorageLocal\" created-in=\"PS-221.5591.58\">\n    <data-source name=\"books_I425@localhost\" uuid=\"70565d18-468d-4197-94d4-880077ac668d\">\n      <database-info product=\"MariaDB\" version=\"10.4.24-MariaDB\" jdbc-version=\"4.2\" driver-name=\"MariaDB Connector/J\" driver-version=\"2.7.3\" dbms=\"MARIADB\" exact-version=\"10.4.24\" exact-driver-version=\"2.7\">\n        <extra-name-characters>#@</extra-name-characters>\n        <identifier-quote-string>`</identifier-quote-string>\n      </database-info>\n      <case-sensitivity plain-identifiers=\"lower\" quoted-identifiers=\"lower\" />\n      <secret-storage>master_key</secret-storage>\n      <user-name>phpuser</user-name>\n      <schema-mapping>\n        <introspection-scope>\n          <node kind=\"schema\" qname=\"@\" />\n        </introspection-scope>\n      </schema-mapping>\n    </data-source>\n  </component>\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/dataSources.local.xml b/.idea/dataSources.local.xml
--- a/.idea/dataSources.local.xml	(revision 5a58435ec921a67b7528d40df86c0251d32463b8)
+++ b/.idea/dataSources.local.xml	(date 1653342527882)
@@ -1,12 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="dataSourceStorageLocal" created-in="PS-221.5591.58">
-    <data-source name="books_I425@localhost" uuid="70565d18-468d-4197-94d4-880077ac668d">
-      <database-info product="MariaDB" version="10.4.24-MariaDB" jdbc-version="4.2" driver-name="MariaDB Connector/J" driver-version="2.7.3" dbms="MARIADB" exact-version="10.4.24" exact-driver-version="2.7">
+    <data-source name="group5_books_db@localhost" uuid="f452442c-e51c-4ee1-b56f-4fd50d500961">
+      <database-info product="MariaDB" version="10.4.21-MariaDB" jdbc-version="4.2" driver-name="MariaDB Connector/J" driver-version="2.7.3" dbms="MARIADB" exact-version="10.4.21" exact-driver-version="2.7">
         <extra-name-characters>#@</extra-name-characters>
         <identifier-quote-string>`</identifier-quote-string>
       </database-info>
-      <case-sensitivity plain-identifiers="lower" quoted-identifiers="lower" />
+      <case-sensitivity plain-identifiers="mixed" quoted-identifiers="mixed" />
       <secret-storage>master_key</secret-storage>
       <user-name>phpuser</user-name>
       <schema-mapping>
